### PR TITLE
Init R package and test classifier filter

### PR DESCRIPTION
**Pull request by alexvpickering**
# Background
#### Link to issue 

[BIOMAGE-1141](https://biomage.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=BIOMAGE&modal=detail&selectedIssue=BIOMAGE-1141)

[BIOMAGE-1127](https://biomage.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=BIOMAGE&modal=detail&selectedIssue=BIOMAGE-1127)

#### Link to staging deployment URL 
https://ui-rstudio.scp-staging.biomage.net/data-management

#### Anything else the reviewers should know about the changes here

- skeleton added for `testthat`
- added tests for classify filter
-  remove uneccessary `library(...) // require(...)`
- task functions renamed so that not all `task` (mainly for code navigation) and moved to top of files.
- mixed usage of `scdata` and `seurat_object` → `scdata`
- styled package with `styler` (Rstudio →Addins → Style active package)
- removed 10x version checking code as filenames are standardized in UI or API (e.g. see `download_cellranger` - we only download `features.tsv.gz`, not `genes.tsv(.gz)`
- consolidated some repetitive code and added timings to QC steps
- changed error handling so that more concise (doesn't print all the error-handling specific nonsense):

before: 
image

after:
image

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [x] Unit tests written
- [x] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [x] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [x] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: pipeline-runner/R/gem2s-1-download_cellranger.R [hunk 1]
```diff
@@ -0,0 +1,59 @@
+download_cellranger <- function(input, pipeline_config) {
```
Comment (ogibson-biomage) [inline pipeline-runner/R/gem2s-1-download_cellranger.R:1]: Are these renamed files equal to the old ones or have you made changes?

File: pipeline-runner/R/gem2s-3-run_emptydrops.R [hunk 2]
```diff
@@ -1,28 +1,52 @@
-################################################
```
Comment (ogibson-biomage) [inline pipeline-runner/R/gem2s-3-run_emptydrops.R:1]: I liked this :(

Comment (alexvpickering) [reply pipeline-runner/R/gem2s-3-run_emptydrops.R:1]: They cause issues with folding in Rstudio. If you type `alt+o` it folds by function. This breaks that.

File: pipeline-runner/R/gem2s-2-load_cellranger.R [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  # We include in variable scdata_list all the sparse matrix per sample
+  message("Creating raw dataframe...")
+  scdata_list <- call_read10x(config)
+
+  # We store the raw scdata_list since for the emptyDrops since to compute the background we cannot remove any cells.
+  message("Exporting raw scdata for emptyDrops...")
+  saveRDS(scdata_list, file = "/output/pre-doublet-scdata_list.rds", compress = FALSE)
+
+  message("Step 1 completed.")
+  print(list.files(paste("/output", sep = "/"), all.files = TRUE, full.names = TRUE, recursive = TRUE))
```
Comment (ogibson-biomage) [inline pipeline-runner/R/gem2s-2-load_cellranger.R:25]:
```suggestion
  message("Step 2 completed.")
  print(list.files(paste("/output", sep = "/"), all.files = TRUE, full.names = TRUE, recursive = TRUE))
```

File: pipeline-runner/R/gem2s-4-score_doublets.R [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+  scdata_list <- scdata_list[samples]
+
+  message("calculating probability of barcodes being doublets...")
+  for (sample_name in names(scdata_list)) {
+    compute_doublet_scores(scdata_list[[sample_name]], sample_name)
+  }
+
+  message("Step 2-2 completed.")
+  print(list.files(paste("/output", sep = "/"), all.files = TRUE, full.names = TRUE, recursive = TRUE))
```
Comment (ogibson-biomage) [inline pipeline-runner/R/gem2s-4-score_doublets.R:28]:
```suggestion
  message("Step 4 completed.")
  print(list.files(paste("/output", sep = "/"), all.files = TRUE, full.names = TRUE, recursive = TRUE))
```

File: pipeline-runner/R/gem2s-6-prepare_experiment.R [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+  # The RJSONIO library add '' to boolean keys, so we will remove them.
+  exportJson <- gsub('\"true\"', "true", exportJson)
+  exportJson <- gsub('\"false\"', "false", exportJson)
+  # Tranform null into []
+  exportJson <- gsub("null", "[]", exportJson)
+  message("config file...")
+  write(exportJson, "/output/config_dataProcessing.json")
+
+  message("Step 4 completed.")
+  print(list.files(paste("/output", sep = "/"), all.files = TRUE, full.names = TRUE, recursive = TRUE))
```
Comment (ogibson-biomage) [inline pipeline-runner/R/gem2s-6-prepare_experiment.R:231]:
```suggestion
  message("Step 6 completed.")
  print(list.files(paste("/output", sep = "/"), all.files = TRUE, full.names = TRUE, recursive = TRUE))
```

**Review by ogibson-biomage — APPROVED** [review]

**Pull request merged**